src/components/AddComment.jsx

```diff
@@ -0,0 +1,85 @@
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import InputGroup from "react-bootstrap/InputGroup";
+import { useContext, useState, useEffect } from "react";
+import { CommentsContext } from "../contexts/CommentsContext";
+import { postComment } from "../api";
+
+export const AddComment = ({ article_id }) => {
+  const [commentInput, setCommentInput] = useState("");
+  const { setComments } = useContext(CommentsContext);
+  const [uploadMessage, setUploadMessage] = useState(null);
+
+  const handleChange = (e) => {
+    setCommentInput(e.target.value);
+    setUploadMessage(null);
+  };
+
+  const uploadComment = (e) => {
+    e.preventDefault();
+
+    if (commentInput === "") {
+      setUploadMessage("Please type a valid comment before posting");
+    } else {
+      setUploadMessage("Your comment is uploading...");
+
+      const newComment = {
+        username: "tickle122",
+        body: commentInput,
+      };
+
+      postComment(article_id, newComment)
+        .then((comment) => {
+          setComments((currComments) => {
+            setUploadMessage("Comment succesfully posted!");
```

Comment:
`setUploadMessage` is setting state in the callback that is being used to set Comments, this can sometimes yield some unexpected behaviours in React